### PR TITLE
Add some extra logging to GCNotificationTests

### PR DESCRIPTION
**Pull request by swgillespie**
I couldn't repro the failure https://github.com/dotnet/corefx/issues/12991 so this PR adds some logging to the failing test so that we can see what failed next time this occurs. Apparently `RemoteInvoke` doesn't show the standard output of the child process upon failure, so all logging is moved to asserts so that the exception message is propagated to the xunit runner.

### CODE REVIEW

**Comment by swgillespie** [comment]
@ramarag PTAL?

File: src/System.Runtime/tests/System/GCTests.netstandard1.7.cs [hunk 1]
```diff
@@ -67,22 +67,23 @@ public static void GCNotificationNegTests()
             Assert.Throws<ArgumentOutOfRangeException>(() => GC.WaitForFullGCComplete(-2));
         }
 
-        [Fact]
+        [Theory]
+        [InlineData(new object[] { true, -1 })]
```
Comment (ramarag) [inline src/System.Runtime/tests/System/GCTests.netstandard1.7.cs]: I am not so sure of changing it to theory, as now for each input a new process is going to spun up, i will leave it up to corefx test maintainers on this descision. @stephentoub please reroute\make a call

**Comment by ramarag** [comment]
LGTM  modulo comment

File: src/System.Runtime/tests/System/GCTests.netstandard1.7.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.Runtime/tests/System/GCTests.netstandard1.7.cs]: It should be fine.  We spin up a lot of processes during test runs.

File: src/System.Runtime/tests/System/GCTests.netstandard1.7.cs [hunk 2]
```diff
@@ -67,22 +67,23 @@ public static void GCNotificationNegTests()
...
+        [InlineData(new object[] { false, -1 })]
+        [InlineData(new object[] { true, 0 })]
+        [InlineData(new object[] { false, 0 })]
+        [InlineData(new object[] { true, 100 })]
+        [InlineData(new object[] { false, 100 })]
+        [InlineData(new object[] { true, int.MaxValue })]
+        [InlineData(new object[] { false, int.MaxValue })]
```
Comment (justinvp) [inline src/System.Runtime/tests/System/GCTests.netstandard1.7.cs]: Nit: The `new object[] { ... }` isn't necessary. They can be simplified as:

```c#
[InlineData(true, -1)]
[InlineData(false, -1)]
[InlineData(true, 0)]
[InlineData(false, 0)]
...
```

**Pull request merged**